adhocteam/Head-Start-TTADP
# Fix url routing

**Description of change**

 * The login button sends the referrer header (note it is actually spelled `referer` in the header, [some backstory](https://stackoverflow.com/questions/3087626/was-the-misspelling-of-the-http-field-name-referer-intentional)) that is used to redirect the user to the proper frontend page after login.
 * Add frontend 404 page (an actual html page)
 * Add backend 404 route (a text response)

**How to test**

1. Without logging in go to `https://tta-smarthub-sandbox.app.cloud.gov/activity-reports` and login. You should be redirected to the activity reports page after login.
2. Go to `https://tta-smarthub-sandbox.app.cloud.gov/test`. You should see a generic 404 page.
3. Go to `https://tta-smarthub-sandbox.app.cloud.gov/api/test`. You should see "Not found" text.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/121

**Checklist**

- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
